### PR TITLE
Upgrade to go 1.7

### PR DESCRIPTION
**Pull request by dmathieu**
* Upgrade go within `Godeps`.
* Move all dependencies to `vendor`.

### CODE REVIEW

File: registry/repositories.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ type repositoriesResponse struct {
 	Repositories []string `json:"repositories"`
 }
 
-func (registry *Registry) Repositories() (repos []string, err error) {
```
Comment (ojacobson) [inline registry/repositories.go]: I think this conflicts with your other PR regarding compound return values. Which one are you preferring, right now?

Comment (dmathieu) [inline registry/repositories.go]: I definitely didn't mean to include that. I missed it in the large number of files updated. Good catch!
I've removed it.

**Comment by dmathieu** [comment]
Travis is failing because there's no `.travis.yml` file. See #15

**Comment by ojacobson** [comment]
Looks good! Thanks for keeping language versions up to date. Merge at will, then go ahead with #15.

**Pull request merged**